File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        m = {}
        n = len(nums)

        for i in range(n):
            comp = target - nums[i]
            if comp in m:
                return [m[comp], i]
            m[nums[i]] = i

        return []
        

File: 1-two-sum/two-sum.go
func twoSum(nums []int, target int) []int{
    numMap := make(map[int]int)
    for i, num := range nums{
        comp := target - num
        if index, found := numMap[comp]; found{
            return []int{index, i}
        }
        numMap[num] = i
    }
    return nil
}